stubs/parsimonious/parsimonious/nodes.pyi

```diff
@@ -0,0 +1,40 @@
+from re import Match
+from typing import Any, Callable, Iterator, NoReturn, TypeVar
+from typing_extensions import Self
+
+from parsimonious.exceptions import UndefinedLabel as UndefinedLabel, VisitationError as VisitationError
+from parsimonious.expressions import Expression as Expression
+from parsimonious.grammar import Grammar as Grammar
+
+class Node:
+    expr: Expression
+    full_text: str
+    start: int
+    end: int
+    children: list[Node]
+    def __init__(self, expr: Expression, full_text: str, start: int, end: int, children: list[Node] | None = ...) -> None: ...
+    @property
+    def expr_name(self) -> str: ...
+    def __iter__(self) -> Iterator[Node]: ...
+    @property
+    def text(self) -> str: ...
+    def prettily(self, error: Node | None = ...) -> str: ...
+
+class RegexNode(Node):
+    match: Match[str]
+
+class RuleDecoratorMeta(type):
+    def __new__(metaclass: type[Self], name: str, bases: tuple[type, ...], namespace: dict[str, Any]) -> Self: ...
```

Comment:
Pyright is complaining about this line:
`error: __new__ override should take a "cls" parameter (reportSelfClsParameterName)`

The error makes sense, but this is how it's written in the source code. `metaclass` gets passed to the `cls` parameter of `type.__new__`.

I tried adding this method to the allowlist but doesn't seem to have suppressed the error?

Reply:
That allowlist is for stubtest, a different tool than pyright. To shut up pyright here you'd have to use `# type: ignore`.

Reply:
Perfect, good to know!